src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,33 @@
      * Андрей - 600
      * София - 900</p>
      */
+
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+            throws IllegalDateParametersException {
+
+        LocalDate fromDate = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate toDate = LocalDate.parse(dateTo, FORMATTER);
+
+        StringBuilder result = new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo);
```

Comment:
let's move this line after throw exception
let's use `append()` instead of `+`

Reply:
done